packages/nimble-components/src/rich-text/models/tests/markdown-serializer.spec.ts

```diff
@@ -1,293 +1,303 @@
+                inputMarkdown: '&',
+                outputMarkdown: '&'
             },
             {
                 name: 'HTML entities <&trade;>',
-                html: '&trade;',
-                markdown: '™'
+                inputMarkdown: '™',
+                outputMarkdown: '™'
             },
             {
                 name: 'HTML entities <&euro>',
-                html: '&euro;',
-                markdown: '€'
+                inputMarkdown: '€',
+                outputMarkdown: '€'
             },
             {
                 name: 'Markdown syntax strings <*>',
-                html: '*',
-                markdown: String.raw`\*`
+                inputMarkdown: '*',
+                outputMarkdown: '* '
             },
             {
                 name: 'Markdown syntax strings <**>',
-                html: '**',
-                markdown: String.raw`\*\*`
+                inputMarkdown: '**',
+                outputMarkdown: r`\*\*`
             },
             {
                 name: 'Markdown syntax strings <_>',
-                html: '_',
-                markdown: String.raw`\_`
-            }
-        ];
-
-        const focused: string[] = [];
-        const disabled: string[] = [];
-        for (const value of supportedNodesMarks) {
-            const specType = getSpecTypeByNamedList(value, focused, disabled);
-            specType(
-                `Should return ${value.name} markdown (${value.markdown}) when its respective node is passed`,
-                // eslint-disable-next-line @typescript-eslint/no-loop-func
-                () => {
-                    const node = getNode(value.html);
-
-                    expect(
-                        RichTextMarkdownSerializer.serializeDOMToMarkdown(node)
-                    ).toBe(value.markdown);
-                }
-            );
-        }
-    });
-
-    describe('various not supported nodes should be serialized to a plain text', () => {
-        const notSupportedNodesMarks: {
-            name: string,
-            html: string,
-            plainText: string
-        }[] = [
+                inputMarkdown: '_',
+                outputMarkdown: r`\_`
+            },
             {
                 name: 'Blockquote',
-                html: '<blockquote><p>Blockquote</p></blockquote>',
-                plainText: 'Blockquote'
+                inputMarkdown: '> blockquote',
+                outputMarkdown: '\\> blockquote'
+            },
+            {
+                name: 'Code',
+                inputMarkdown: '`code`',
+                outputMarkdown: '\\`code\\`'
             },
-            { name: 'Code', html: '<code>Code</code>', plainText: 'Code' },
             {
                 name: 'CodeBlock',
-                html: '<pre><code>CodeBlock</code></pre>',
-                plainText: 'CodeBlock'
+                inputMarkdown: '```\nCodeBlock\n```',
+                outputMarkdown: '\\`\\`\\` CodeBlock \\`\\`\\`'
```

Comment:
For all these tests where the output has added characters, it would be good to have additional tests that use the `outputMarkdown` as inputs. We should verify that re-loading the resulting markdown doesn't keep adding characters every time.

Reply:
Added testcases to check the character addition markdown output when reloading.